libraries/core/lib/metaheuristics/SearchAlgorithm.ts

```diff
@@ -148,7 +148,7 @@ export abstract class SearchAlgorithm<T extends Encoding> {
       );
 
       // Start next iteration of the search process
-      this._iterate(budgetManager, terminationManager);
+      await this._iterate(budgetManager, terminationManager);
```

Comment:
Why is this necessary?

Reply:
Because iterate is an async function (in the implementation in EvolutionaryAlgorithm.ts) and otherwise we would not wait for the function to finish.

Reply:
Wouldn't this have caused problems in the past or has it only recently been modified to be async?

Reply:
No, I think I accidentally removed it in one of the refactors, because I remember it used to await correctly. I just added it back now.